### ISSUE TITLE
[Accessibility] Invalid role='navigation' being added to <ul> when fd-list-link directive is used for <a> inside <li>

### ISSUE THREAD

**Issue by harshitha-s-kanchan** [SAP/fundamental-ngx#11592]
#### Is this a bug, enhancement, or feature request?
bug

#### Describe your proposal.
role=navigation should not be used on `<ul>` tag as its invalid role for ul tag, and throws error in Access Assistant tool.

Additionally, JAWS screenreader does not read out the count of list items when role=navigation is added to `<ul>`

#### Can you handle that on the application side
No, as the role is getting added in fd-list component

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
Angular - 15
NGX - 0.43.21 (Issue also present in the latest version)
component - `fd-list`, `fd-list-link`
Screen reader - JAWS 2023

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
On inspecting the UL in Access Assistant tool, below are the issues found,
<img width="1710" alt="image" src="https://github.com/SAP/fundamental-ngx/assets/149040487/22b8866d-2c53-44bd-be88-f7a2e598eba4">
<img width="1710" alt="image" src="https://github.com/SAP/fundamental-ngx/assets/149040487/e56f0571-2c10-492c-9804-dfb440b014d9">

<img width="1713" alt="image" src="https://github.com/SAP/fundamental-ngx/assets/149040487/06889493-1fba-491e-895a-bb21cf5146f1">

#### In case this is Accessibility related topic, did you consult with an accessibility expert? If not, please do so and share their recommendations. 
Yes the issue is raised by Central Accessibility team. 

Below are the two issues mentioned by them

"_Ensure ARIA roles, states, and properties are valid_"
"_The role attribute value given to this UL is not allowed.  This element's role attribute can be set to one of the following text values: directory | group | listbox | menu | menubar | radiogroup | tablist | toolbar | tree; or the role attribute can be removed._"

"_Ensure list items are found in a list container_"
"_This LI does not have a ul element (without an ARIA-assigned role); ol element (without an ARIA-assigned role); an element with a role set to the value: list as a parent; or a ul element (without an ARIA-assigned role), ol element (without an ARIA-assigned role) or element with a role set to the value 'list' with an aria-owns attribute set to the ID of the element in the same DOM_"

**Comment by droshev** [comment]
@nikolay-kolarov  what would you advise us in this scenario?

**Comment by nikolay-kolarov** [comment]
Hi @droshev,

The problem is that the role navigation invalidates the semantic of the UL and then the listitems, do not have a parent list item.

In my opinion, if you want to keep the navigation semantic, it needs to wrap the ul.

E.g. : <nav aria-label="...." ...><ul role="tree" ...</ul></nav>.

Best Regards,
Nikolay
